### PR TITLE
fix: focus management for stepper

### PR DESCRIPTION
**Pull request by coryrylan**
## PR Checklist

Please check if your PR fulfills the following requirements:

* [x] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] If applicable, have a visual design approval

## PR Type

What kind of change does this PR introduce?

* [x] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [ ] Documentation content changes
* [ ] clarity.design website / infrastructure changes
* [ ] Other... Please describe:

## What is the current behavior?

When navigating between steps using the next button the focus was not set on the next step heading button.

Issue Number: N/A
https://github.com/vmware/clarity/issues/3528

## What is the new behavior?
When the user clicks next on a valid step the next step is automatically focused.

## Does this PR introduce a breaking change?

* [ ] Yes
* [x] No

### CODE REVIEW

**Comment by coryrylan** [comment]
WIP: need to update the aria announcements to remove the success state as it is no longer needed with the focus management.

File: src/clr-angular/accordion/accordion-panel.html [hunk 1]
```diff
@@ -11,12 +10,17 @@
         class="clr-accordion-header-button"
         (click)="togglePanel()"
         [id]="getAccordionHeaderId(panel.templateId)"
-        [disabled]="panel.disabled"
+        [disabled]="isAccordion && panel.disabled"
+        [attr.aria-disabled]="!isAccordion && panel.disabled"
```
Comment (jeeyun) [inline src/clr-angular/accordion/accordion-panel.html]: Can you tell me why accordion uses `disabled` attribute and stepper uses `aria-disabled`? Also doesn't presence of `disabled` attribute make it disabled by default even if its value is set to `false`?

Comment (coryrylan) [reply src/clr-angular/accordion/accordion-panel.html]: The reason the accordion uses `disabled` is we want the button truly disabled, cant be focused or clicked. When using `disabled` you can't trigger a focus so on the stepper we use `aria-disabled` so we can auto focus to the next step and read both the step title and let the screen user know the button is disabled.

When `panel.disabled` is `false` that will set the `[disabled]` JS property on the button which will reflect that state into the HTML and remove the `disabled` attribute. If I had used `[attr.disabled]="panel.disabled"` then I set the attribute skipping the property completely and `disabled="false"` would be added to the button disabling it incorrectly.

Comment (jeeyun) [reply src/clr-angular/accordion/accordion-panel.html]: Thanks for the detailed explanation. It all makes sense now. 👍

**Pull request merged**